enarx-keep-sgx/src/component.rs

```diff
@@ -91,10 +91,22 @@ pub struct Component {
 impl<'a> Component {
     /// Loads a binary from a file
     pub fn from_path(path: impl AsRef<Path>) -> Result<Self> {
-        let data = std::fs::read(path)?;
+        let file = std::fs::File::open(path)?;
+        let len = file.metadata()?.len() as usize;
+        let (data, _mmap) = unsafe {
+            let addr = mmap::map(0, len, libc::PROT_READ, libc::MAP_PRIVATE, Some(&file), 0)?;
+            let data = std::slice::from_raw_parts(addr as *const u8, len);
+            (
+                data,
+                mmap::Unmap::new(Span {
+                    start: addr,
+                    count: len,
+                }),
+            )
+        };
```

Comment:
Just a style nit: how about something like this to make it easier to follow the braces inside the tuple parens:

```diff
         let (data, _mmap) = unsafe {
             let addr = mmap::map(0, len, libc::PROT_READ, libc::MAP_PRIVATE, Some(&file), 0)?;
             let data = std::slice::from_raw_parts(addr as *const u8, len);
-            (
-                data,
-                mmap::Unmap::new(Span {
-                    start: addr,
-                    count: len,
-                }),
-            )
+            let unmap = mmap::Unmap::new(Span {
+                start: addr,
+                count: len,
+            });
+            (data, unmap)
         };
 
         // Parse the file.
```

Thoughts?

Reply:
Thanks for your advice! I think it's better for readability.